preprocess_observations/preprocess_adusumilli_melt.py

```diff
@@ -0,0 +1,328 @@
+    """
+    Remap Adusumilli et al. (2020) melt rates to 10 km comparison grid
+
+    Parameters
+    ----------
+    in_filename : str
+        The Adusumilli et al. (2020) melt rates in NetCDF format
+
+    out_prefix : str
+        A prefix for the file to contain the Adusumilli et al. (2020) melt
+        rates and melt fluxes remapped to the comparison grid
+
+    date : str
+        A date string to append to the file name.
+
+    task_count : int
+        The number of MPI tasks to use to create the mapping file
+    """
+    ds = xr.open_dataset(in_filename)
+
+    melt_attrs = ds.meltRate.attrs
+    uncert_attrs = ds.meltRateUncertainty.attrs
+
+    mask = ds.meltRate.notnull()
+    ds['meltRate'] = ds.meltRate.where(mask, 0.)
+    ds['meltMask'] = mask.astype(float)
+    mask = ds.meltRateUncertainty.notnull()
+    ds['meltRateUncertSqr'] = (ds.meltRateUncertainty**2).where(mask, 0.)
+    ds['uncertMask'] = mask.astype(float)
+    ds = ds.drop_vars(['lat', 'lon', 'meltRateUncertainty'])
+
+    in_x = ds.x.values
+    in_y = ds.y.values
+    lx = np.abs(1e-3 * (in_x[-1] - in_x[0]))
+    ly = np.abs(1e-3 * (in_y[-1] - in_y[0]))
+    dx = np.abs(1e-3 * (in_x[1] - in_x[0]))
+
+    in_grid_name = f'{lx:g}x{ly:g}km_{dx:g}km_Antarctic_stereo'
+
+    in_projection = pyproj.Proj('+proj=stere +lat_ts=-71.0 +lat_0=-90 '
+                                '+lon_0=0.0 +k_0=1.0 +x_0=0.0 +y_0=0.0 '
+                                '+ellps=WGS84')
+
+    in_descriptor = ProjectionGridDescriptor.create(
+        in_projection, in_x, in_y, in_grid_name)
+
+    width = 6000.
+    reses = [1., 4., 10.]
+
+    for res in reses:
+        x_max = 0.5 * width * 1e3
+        nx = int(width / res) + 1
+        out_x = np.linspace(-x_max, x_max, nx)
+
+        out_grid_name = f'{width:g}x{width:g}km_{res:g}km_Antarctic_stereo'
+
+        out_projection = get_antarctic_stereographic_projection()
+
+        out_descriptor = ProjectionGridDescriptor.create(
+            out_projection, out_x, out_x, out_grid_name)
+
+        method = 'conserve'
+
+        map_filename = f'map_{in_grid_name}_to_{out_grid_name}_{method}.nc'
+
+        remapper = Remapper(in_descriptor, out_descriptor, map_filename)
+
+        if not os.path.exists(map_filename):
+            remapper.build_mapping_file(method=method, mpiTasks=task_count,
+                                        esmf_parallel_exec='srun')
+
+        ds_out = remapper.remap(ds)
+        mask = ds_out.meltMask > 0.
+        ds_out['meltRate'] = ds_out.meltRate.where(mask)
+        ds_out.meltRate.attrs = melt_attrs
+        mask = ds_out.uncertMask > 0.
+        ds_out['meltRateUncertainty'] = \
+            (np.sqrt(ds_out.meltRateUncertSqr)).where(mask)
+        ds_out.meltRateUncertainty.attrs = uncert_attrs
+        ds_out = ds_out.drop_vars(['meltRateUncertSqr'])
+        ds_out.to_netcdf(f'{out_prefix}_{out_grid_name}.{date}.nc')
+
+
+def compute_regional_means(in_filename, out_prefix, date, chunk_size=50000,
```

Comment:
@cbegeman, here's the function I would appreciate you sanity-checking if you have time.